### PR TITLE
Add info in kubectl get triggerauthentication

### PR DESCRIPTION
**Pull request by SpiritZhou**
Add scaledobjetc/socaledjob info when kubectl get triggerauthentication/clustertriggerauthentication.
Here is more info: https://github.com/kedacore/keda/issues/796 

Example:
image

### Checklist

- [N/A] When introducing a new scaler, I agree with the [scaling governance policy](https://github.com/kedacore/governance/blob/main/SCALERS.md)
- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [N/A] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [N/A] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes https://github.com/kedacore/keda/issues/796

### CODE REVIEW

File: CHANGELOG.md
Comment (tomkerkhove) [inline CHANGELOG.md:56]:
```suggestion
- **General:**: Add ScaledObject/ScaledJob names to output of `kubectl get triggerauthentication/clustertriggerauthentication` ([#796](https://github.com/kedacore/keda/issues/796))
```

**Review by tomkerkhove — COMMENTED** [review]
As discussed, would you mind adding some tests + example output please?

File: apis/keda/v1alpha1/triggerauthentication_types.go
Comment (zroubalik) [inline apis/keda/v1alpha1/triggerauthentication_types.go:94]: Please rename `allscaledobjectnamesstring` to something a little bit shorter.  Maybe  just `scaledObjects` or `usedByScaledObjects`? Either way please use camelCase.

File: apis/keda/v1alpha1/triggerauthentication_types.go
Comment (zroubalik) [inline apis/keda/v1alpha1/triggerauthentication_types.go:288]: I am not sure this is the right approach. Wouldn't be better to use methods on `TriggerAuthenticationStatus` which do everything in one step? Something like `AddScaledObjectName(soName string)`/`RemoveScaledObjectName(soName string)`. It would do all the magic inside (`kedacontrollerutil.AppendIntoString`...). These methods could probably return just a new Status that should be then applied to the cluster (please use `client.Patch()` isntead of  `Update()`). You can check how do we update Status for ScaleObjects: https://github.com/kedacore/keda/blob/main/pkg/util/status.go

File: controllers/keda/util/string_lists.go
Comment (zroubalik) [inline controllers/keda/util/string_lists.go:43]:
```suggestion
// AppendIntoString appends a new string into a string that has seprator
```

In Go, comments on functions should start with the name of the function, please check that also in the rest of the PR.

**Review by zroubalik — COMMENTED** [review]
Looking good, we should def add a tests for this.

File: apis/keda/v1alpha1/triggerauthentication_types.go
Comment (SpiritZhou) [reply apis/keda/v1alpha1/triggerauthentication_types.go]: Rename allscaledobjectnamesstring to scaledObjects

**Comment by SpiritZhou** [comment]
It seems like adding infrastructure test should provide a new PR? I will add a PR about the test after PR merging.  @tomkerkhove @zroubalik

**Comment by tomkerkhove** [comment]
It depends what you need. Our cluster is managed in another PR but you should be good for this PR.

For e2e tests we deploy resources to the cluster which is part of the e2e rest itself so we can do it in this PR.

**Comment by SpiritZhou** [comment]
Thanks, @tomkerkhove. I added a e2e test in the PR.

File: controllers/keda/triggerauthentication_controller.go
Comment (zroubalik) [inline controllers/keda/triggerauthentication_controller.go:148]: both these funcitons are called only once in the `UpdateTriggerAuthenticationStatusHandler`, I don't think it is necessary to create a new function for each (even public ones). I think you can simplify and merge the code to query both TriggerAuth and TriggerAuth Status in the single function or command `UpdateTriggerAuthenticationStatusHandler`

File: controllers/keda/triggerauthentication_controller.go
Comment (zroubalik) [inline controllers/keda/triggerauthentication_controller.go:159]: I don't think this should be in the TA_controller code, as it is not used here at all. I think that it would be better to create a new package `pkg/status` and move this logic there. (also move there stuff from `pkg/util/status.go`)

File: controllers/keda/triggerauthentication_controller.go
Comment (zroubalik) [inline controllers/keda/triggerauthentication_controller.go:180]: this error imho cannot happen as you query the status like this:
```go
case *kedav1alpha1.TriggerAuthentication:
		return &obj.Status, nil
```

File: pkg/util/status.go
Comment (zroubalik) [inline pkg/util/status.go:66]: This is called only once from a single place in the code base. Think if we need to have it this way and having this as a public method.

File: tests/internals/update_ta/update_ta_test.go
Comment (zroubalik) [inline tests/internals/update_ta/update_ta_test.go:310]: Could you please make these functions more generic (to accept name, namespace ... as parameters) and move them to https://github.com/kedacore/keda/blob/main/tests/helper/helper.go as it might be beneficial for other following tests

**Review by zroubalik — COMMENTED** [review]
Would be nice to add a functional test here: https://github.com/kedacore/keda/blob/93851e73c7710bbf815e2fd200dbe2e3b79bf5a8/controllers/keda/scaledobject_controller_test.go#L235

(create SO _ check TA status, remove SO check TA status again, create 2 SOs check TA).

And also create a separate test for scaledjob_controller that is testing the same.

File: pkg/status/status.go
Comment (zroubalik) [inline pkg/status/status.go:68]: could you please add a short doc comment here on what this function does and returns?

File: pkg/status/status.go
Comment (zroubalik) [inline pkg/status/status.go:91]: could you please add a short doc comment here on what this function does and returns?

File: tests/helper/helper.go
Comment (zroubalik) [inline tests/helper/helper.go:800]: could you please add a short doc comment here on what this function does and returns?

File: controllers/keda/triggerauthentication_controller.go
Comment (zroubalik) [inline controllers/keda/triggerauthentication_controller.go:24]: please remove this change

**Review by zroubalik — COMMENTED** [review]
Looking good, just a few nits to be done. In general it is better to add comments on every functions that do something non-trivial.

File: controllers/keda/triggerauthentication_controller.go
Comment (tomkerkhove) [reply controllers/keda/triggerauthentication_controller.go:24]:
```suggestion
	"k8s.io/apimachinery/pkg/api/errors"
```

**Comment by tomkerkhove** [comment]
LGTM, but I'll leave it up to @zroubalik

**Review by zroubalik — APPROVED** [review]
LGTM


Thanks!

**Pull request merged**
